### ISSUE TITLE
Error: pier: serf unexpectedly shut down

### ISSUE THREAD

**Issue by dentropy** [mopfel-winrux/urbit-umbrel#26]
I get this error when booting my planet on [umbrel](https://github.com/getumbrel/umbrel-apps/tree/master/urbit) . I have run the following command and got the sane error,

``` bash
sudo docker exec -it urbit_manager_1 bash
bash /app/boot_pier.sh tartem-milteb 34343 &
```

I researched how to troubleshoot this and some issues said to downgrade. I can't seem to downgrade apps on umbrel so I am out of luck.

**Comment by dentropy** [comment]
I found the solution to my problem,


Source: [Urbit Troubleshooting](https://galactictribune.net/post/urbit-troubleshooting)

=======

pier: serf unexpectedly shut down


This means we ran out of memory on the loom (the persistent memory arena in urbit's runtime, currently limited to 2GB). This is currently always handled as a fatal error -- we're working on tooling to automatically recover where possible.

These are errors are sometimes ephemeral, since the loom holds both are persistent state and the "workspace" for processing a given event. In which case, restarting urbit is all that's required. But they usually recur, and require further intervention to be stopped.

Troubleshooting steps for solving this problem:

Run |pack (see below) - defragments the persistent state
Run |meld (see below) - is a global deduplicator: it reallocates all persistent state off the loom, unifying any duplicate nouns it discovers along the way, and then copies them back to the loom.
